history.go

```diff
@@ -59,14 +68,19 @@ func (rh *resultHistory) List() []*result {
 	rh.mu.Lock()
 	defer rh.mu.Unlock()
 
-	return rh.results[:]
+	return append(rh.preservedFailedResults[:], rh.results...)
```

Comment:
You need to de-dupe here

Reply:
Not true, actually. Maybe there is confusion about how I implemented this. I am adding results to the `preservedFailedResults` slice only after they expire from the main `results` slice. As a result, no result will be in both at the same time. 

If we put failed results into this slice immediately, then I think it would make more sense to have one list for successes and one list for failures, then merge the two on `List`.

Reply:
Ah, I see what you're doing now. A brief comment would help.